enhancements/sig-policy/72-content-to-ansilbe-job/README.md

```diff
@@ -0,0 +1,188 @@
+  "policy_violation_context" : {
+    "cluster1": {
+      "compliant": "NonCompliant",
+      "details": [
+        {
+          "compliant": "NonCompliant",
+          "history": [
+            {
+            "lastTimestamp": "2022-09-08T16:03:23Z",
+            "message": "NonCompliant; violation - pods not found: [testpod] in namespace default missing"
+            },
+          ],
+        },
+      ]
+    },
+    "cluster2": {
+      ...
+    },
+  },
+}
+```
+
+- `policy_name` is the name of the non-compliance policy that triggers the ansible job.
+ `violation_message` is the latest policy violation message. 
+ They are from the root policy on the hub cluster. 
+ The field path of `policy_name` is `policy.metadata.name`.
+ The field path of `violation_message` is `policy.policyStatus.details[0].history[0].message`
+
+- `namespace` is the namespace of the root policy, which is from `rootPolicy.GetNamespace()`
+
+- `hubcluster` is the name of the hub cluster, which is from `rootPolicy.GetClusterName()`
+
+- `policy_violation_context` is a map. The keys are non-compliant cluster names, and the value is the raw replicated 
+ policy status for each non-compliant cluster.
+ The policy automation looks at the `cluster namespaces` of the hub cluster and gets the replicated policies. 
+ Then it filters out the non-compliant replicated policies and puts the status into the `policy_violation_context` map.
+
+ Suppose the policy automation runs on the hub cluster without the managed cluster. In that case, `policy_violation_context` 
+ would be an empty map.
+ 
+ The field path of the raw status is `policy.policyStatus`. The below unnecessary fields in `policy.policyStatus` will be \
+ skipped and not in `policy_violation_context`:
+ 1. `policy.policyStatus.details[n].templateMeta`.
+ 2. `policy.policyStatus.details[n].history[n].eventName`
+
+### User Stories
+
+#### Story 1
+
+As an OCM Policy add-on user, I would like to get the additional status in an Ansible job's `extra_vars` field. For example,
+regarding the CertificatePolicy policy, which checks for expiring certificates in secrets, I want to get the replicated policy 
+status for each non-compliant cluster so I can know which secret expires on a specific cluster in the Anisble Job. Then, the 
+Ansible Automation Platform can create a notice ticket and provide the violation details. 
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigation
+
+Since `policy_violation_context` is the replicated policy status for all non-compliant managed clusters, the length of 
+`policy_violation_context` may be oversized when the amount of managed clusters is large.
+
+To solve this, we will only pass the top `N` replicated policy status into `policy_violation_context`, where `N` is configurable
+via an optional policy automation field called `policyViolationContextLimit`. If this field is not set, `N` will default to the
+first 1000 managed clusters.
```

Comment:
In large clusters, will the `policy_violation_context` impact performance when N is 1000 ? e.g  etcd, cpu

Reply:
We haven't tested which `N` will cause the performance issue, we can adjust `policyViolationContextLimit` in futrue